### ISSUE TITLE
<dtml-sqltest> not working in Zope 4

### ISSUE THREAD

**Issue by mvoppen** [zopefoundation/Products.ZSQLMethods#14]
I'm currently porting a Zope App from Zope 2.13 to Zope 4.1.1.
There is a simple query working in Zope 2.13 against the same DB (MariaDB 10.3) which causes an exception in Zope 4.1.1

```
select   *
from     counter
<dtml-sqlgroup where>
         <dtml-sqltest public optional type="nb">
</dtml-sqlgroup>
order by url
```
It gives an _ProgrammingError: (1064, 'You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near \'"\'Y\'"\norder by url LIMIT 1000\' at line 4')_
what becomes clear when seeing the generated SQL in Server Log:

```
select   *
from     counter
where
public = b"'Y'"
order by url LIMIT 1000
```

Hardcoding `where` instead of `<dtml-sqlgroup>` makes no difference while
`public = <dtml-var public type='nb'>
`
works well.

**Comment by icemac** [comment]
According to your example the problem seem to be that your `public` variable is of type `bytes` which is not converted to `str` in `sqltest`.

A pull request to fix this is welcome.

**Comment by dataflake** [comment]
Fixed by b18e31694db56107d58777c2bd1f289e2f4e05f6
